distribution.pm

```diff
@@ -141,8 +141,9 @@ sub script_run {
     }
     testapi::type_string "$cmd";
     if ($args{timeout} > 0) {
-        my $str    = testapi::hashed_string("SR" . $cmd . $args{timeout});
-        my $marker = "; echo $str-\$?-" . ($args{output} ? "Comment: $args{output}" : '');
+        my $terminator = ((substr $cmd, -1) eq '&') ? '' : ';';
```

Comment:
1. This breaks if the command is `foo & ` (note the space in the end which could also be a tab or similar).
2. This also breaks if the `&` is escaped, e.g. in `cat foo\&` where the `&` is part of the argument.
3. This breaks if the command ends with a comment although this was broken before of course.

Maybe there are more problems with this. Considering that having code which modifies a shell script is messy anyways I suppose this is kind of ok, though as it does not make things worse.

Reply:
The detection of the termination still breaks in these three cases. I would at least fix case 2. because that would be a valid use which should not be prevented. If that's too much work I'm wondering whether it makes sense to have this kind of validation at all.